internal/controller/streams.go

```diff
@@ -498,15 +516,26 @@ func ConnectStreams(ctx context.Context, logger *zap.SugaredLogger, envMap Envir
 				ciliumDone = nil
 				go manageStream(logger, connectAndStreamFalcoNetworkFlows, sm, falcoDone)
 			}
+
+			// Block until one of the streams fail. Then we will jump to the top of
+			// this loop & try again: authenticate and open the streams.
 			select {
 			case <-ciliumDone:
+				failureReason = "Cilium network flow stream closed"
 			case <-falcoDone:
+				failureReason = "Falco network flow stream closed"
 			case <-resourceDone:
+				failureReason = "Resource stream closed"
 			case <-logDone:
+				failureReason = "Log stream closed"
 			}
 			authConContextCancel()
-			logger.Warn("One or more streams have been closed; closing and reopening the connection to CloudSecure")
 		}
+
+		logger.With(
+			"failureReason", failureReason,
+			"attempt", attempt,
+		).Warn("One or more streams have been closed; closing and reopening the connection to CloudSecure")
```

Comment:
Since the logger with those 2 extra fields is not reused after this log statement, it would be more efficient to just pass them to `Warn`:
```suggestion
		logger.Warn("One or more streams have been closed; closing and reopening the connection to CloudSecure", zap.String("failureReason", failureReason), zap.Int("attempt", attempt))
```
I won't block the PR for this, though. So feel free to just resolve this conversation if you want.